### ISSUE TITLE
Add Disease factor (similar to ontogeny factor) parameter to the Initial Concentration formula

### ISSUE THREAD

**Issue by msevestre** [Open-Systems-Pharmacology/PK-Sim#2018]
I think we need to add another scaling factor. Default value is 1 to make disease modelling for say Liver disease or even CKD v2 possible.

We know that some of this diseases influence the activity of know proteins such as CYP3A4. We could have the disease state set this value to  <1 for less activity or >1 for more.

Of course the problem is that we now have the whole enzyme stuff outside of the individual in Expression Profile. We need to define this parameter at the Individual level (maybe there is the parameter in the ExpressionProfile but it is always the default 1 in that case)

This value would be the only one in the Expression Profile not synchronized automatically with the individual (e.g. we need a special case for this)

Alternatively we force the user to create different profiles... not ideal

What do you think? @Yuri05 @PavelBal @StephanSchaller @JanSchlender @sfrechen 

Note: This is required to implement Liver cirrhosis disease state

**Comment by PavelBal** [comment]
Disease factor as an individual parameter, yes. In theory, the user can combine different expression profiles with a disease state. E.g., extensive metabolizer vs. poor metabolizer expression profiles, then having a disease on top of it would result in different values.

**Comment by msevestre** [comment]
Yes

Now 
```
Parameter: S|Organism|VenousBlood|Plasma|CYP3A4|Initial concentration
	IsFixed: False
	Value: 0 [µmol/l]
	Formula: (1 - HCT) > 0 ? RC * (rel_exp_pls + rel_exp_bc * f_exp_bc_membrane * HCT / (1 - HCT)) : 0
		f_exp_bc_membrane: S|CYP3A4|Fraction expressed in blood cells membrane(1)
		RC: S|CYP3A4|Reference concentration(4.32 [µmol/l])
		rel_exp_bc: S|CYP3A4|Relative expression in blood cells(0)
		rel_exp_pls: S|CYP3A4|Relative expression in plasma(0)
		HCT: S|Organism|Hematocrit(0.47)

Parameter: S|Organism|VenousBlood|Plasma|CYP3A4|Start amount
	IsFixed: False
	Value: 0 [µmol]
	Formula: C_0 * f_onto * V
		C_0: S|Organism|VenousBlood|Plasma|CYP3A4|Initial concentration(0 [µmol/l])
		f_onto: S|Organism|VenousBlood|Plasma|CYP3A4|Ontogeny factor(1)
		V: S|Organism|VenousBlood|Plasma|Volume(0.510932948399336 [l])

```

Why is f_onto defined in the start amount and not initial concentration ?
Where should we define f_disease?

f_onto is a local parameter equal to a global one (just in case) because we have diffent values for liver and guts. We onluy need one here

so do we say maybe start amount = f_onto (local) x f_disease (global) x InitConc x V ? 

or do we rewrite so that 
start_amount = InitCOnc * V 

and we move the factos somewhere else (Init Conc for instance)

**Comment by Yuri05** [comment]
Don't see why we should introduce something like this. 
Thought **different expression profiles** ("CYPxy-Healthy", "CYPxy-CKD", "CYPxy-Liver cirrhosis") must be created for exactly such a use case; where e.g. the "Reference concentration" of the "diseased" expression profile would be adjusted accordingly.

**Comment by msevestre** [comment]
Here you are mixing maybe phénotypes (poor meta , extensive meta etc..) with disease state. You can have both this duplicating information
The ckd disease state will change the ref conc of certain enzymes....how should the user know how to do this?

**Comment by StephanSchaller** [comment]
> The ckd disease state will change the ref conc of certain enzymes....how should the user know how to do this?

But this is not part of the current CKD population, yet, correct? (but may be considered in the future)

**Comment by Yuri05** [comment]
>The ckd disease state will change the ref conc of certain enzymes....how should the user know how to do this?

I guess such a _desease scaling factor_ is **protein dependent**? If so: it should be stored as part of a protein expression BB.
E.g. in a protein expression BB we could add a disease state as **an additional input** and then adjust the default value of the reference concentration (I would not introduce additional scaling factor as explicit parameter).

If such desease scaling factor is **protein independent**: it should become part of the individual BB and then indeed I would just extend the initial amount formula with this new factor.

**Comment by msevestre** [comment]
> If so: it should be stored as part of a protein expression BB.

Well ref concentration is not stored in the protein expression db at the moment. Also we only interact with the expression  DB from the expression profile which is disease state independent

I believe we would want to use the SAME Expression profile to model different disease state by specifying the disease state in the individual using it. 

I still believe that this flag belongs at the individual level and IS protein dependent. It should also be usable for protein not loaded from the expression db.

**Comment by Yuri05** [comment]
>Well ref concentration is not stored in the protein expression db 

"protein expression BB" = "Protein Expression Building Block", not the database :wink: 

>I still believe that this flag belongs at the individual level and IS protein dependent.

So you suggest then, that we would have something like a lookup table below (e.g. in the PK-Sim DB)?
And when adding a protein to individual, PK-Sim would check if the protein name is in the table for the current disease state and set the value of the scale factor in the individual accordingly?
Then indeed we could use the same expression profile for different health states. 
Disease|Protein|Reference concentration scale factor
-|-|-
CKD|Protein 1|0.5
CKD|Protein 2|0.7
...|...|...
Liver cirrhosis|Protein 2|0.4
Liver cirrhosis|Protein 3|0.8
...|...|...

**Comment by msevestre** [comment]
Ok no we have a misunderstanding here. You talked about `protein expression` and I thought you meant the protein expression DB. But in fact you meant the `expression profile`

I do not want to add anything in the DB, What I want to have is the possibility to set a parameter (very similar) to what you have in your table above from the CODE that is associated with each disease state.

So I  have Liver cirrhosis disease state and use Protein2, then the factor will be set to 0.4 from the LiverCirrhosisImplementation class. It does not make sense to put these parameters into the DB as they are disease state specific and we may have different values based on different publications

so in short: what you describe above is exactly it but not comign from a DB. it will come from the implementation of the disease state. Does it make sense

I will define thise disease_factor in the `StartAmountFormula = InitConc * onto_factor * disease_state_factor * volume`

Sounds good?

**Comment by Yuri05** [comment]
>Sounds good? 

Almost :) Can we **adjust (multiply by the factor) the reference concentration value of a protein ** when adding it to the "deseased" individual/population instead of introducing additional scaling factor?

**Comment by msevestre** [comment]
No because we need to ensure the auto update from expression profile => individual

What we could do is introduce a new Parameter

Scaling_factor as a formula
Ana scaling_factor = onto_factor* disease_ factor etc..

That way if we ever have to add yet another factor, we can just change the formula of the factor and that's it
